### DISCUSSION TITLE
BoxBorders around multiindex dataframe

### DISCUSSION THREAD

**Discussion by urosdigital** [DeepSpace2/StyleFrame#151] [Q&A] [answered]
I have asked the same question on StackOverflow so i'll just past the link to it here -> https://stackoverflow.com/questions/73662359/how-to-wrap-multi-index-dataframe-with-box-borders

If someone can help me here...?

**Comment by DeepSpace2** [comment]
Hi, StyleFrame doesn't currently support MultiIndexed dataframes. I have started to implement it but got into a small issue due to the way pandas handle objects that are used in the MultiIndex.

**Comment by urosdigital** [comment]
Hi @DeepSpace2 you wrote "small issue", so we can expect this to be implemented in future, i understand this is not a showstopper?

**Reply by urosdigital** [reply]
Hello @DeepSpace2 any progress on this topic?

**Reply by DeepSpace2** [reply]
@urosdigital Hi, thanks for still showing interest in this. The "small" issue turned out to bigger issue due to the way `pd.MultiIndex` treats mutable objects. It uses an internal hash/cache, so the same object in the multiindex only exists once.

This is problematic for `styleframe` because each "cell" in the dataframe is an object with a value and style. Due to `pd.MultiIndex` caching what it believes to be similar objects, accessing and styling individual indexes/rows is not possible and styles are applied to the wrong rows.

I haven't figured a smart way around it.

**Reply by urosdigital** [reply]
Hi @DeepSpace2 thanks for an update.

**Comment by DeepSpace2** [comment, marked as answer]
Related to https://github.com/DeepSpace2/StyleFrame/issues/129
